### ISSUE TITLE
Disable Redirect on Server Disconnect Config Option 

### ISSUE THREAD

**Issue by ChiboYen** [PaperMC/Velocity#1410]
### Requested Feature

The try section is for both initial connections to the server and also if a player gets kicked, this request is to make it so that the try section is just for initial connections, and all backend server disconnects do not redirect the player. When players get kicked from any server, they should be kicked from the proxy as well under this configurable option.

### Why is this needed?

Queue systems are currently buggy and the implementation is weird on Velocity. The backend server disconnects will simply boot the players into the queue server, resulting in a reconnect loop that cannot be broken. An option to not redirect players after initial connection to the proxy will solve this.

### Alternative Solutions

I've tried contacting plugin developers and getting something made on the plugin-side. This was not possible or successful.

**Comment by Timongcraft** [comment]
I believe that this already exists.  
In [the `advanced` section of the config](https://docs.papermc.io/velocity/configuration#advanced-section) under `failover-on-unexpected-server-disconnect`.  
Also, if properly implemented, queue systems should have no issues overriding these defaults regardless of the value of this config option, and for the best support, you would need an external message broker to detect outages of backend servers, regardless of whether it is Velocity or not.

**Comment by ChiboYen** [comment]
That only applies to `unexpected` server disconnects. Regular commands such as `/miencraft:kick <player>` don't currently fall under that and aren't affected by that config option. We aren't detecting outages of backend servers, we are simply disabling the velocity behavior of reconnecting them to a fallback server.

**Comment by Timongcraft** [comment]
I guess it makes sense then, but I don't understand your reasoning, as Queue Plugins should have no problem circumventing this.
That said, if this option is turned on, plugins cannot stop the proxy from disconnecting the player, or is that intended?

**Comment by ChiboYen** [comment]
From my understanding, queue plugins have no way of knowing if a player got kicked from the backend. If this config option is turned on, plugins can still stop the proxy from disconnecting the player (and instead redirecting them) since such plugins already exist (KickRedirect). I'm just suggesting this because while redirect behavior plugins exist, there is no existing solution to disable the velocity redirect behavior for ALL kicks.

**Comment by Timongcraft** [comment]
No, plugins can already detect kicks with the [`KickedFromServerEvent`](https://jd.papermc.io/velocity/3.3.0/com/velocitypowered/api/event/player/KickedFromServerEvent.html), that's how KickRedirect works. And if you know that plugins like KickRedirect exist, I don't get how plugins should be unable to disconnect the player themselves. But if the proxy were to disconnect the player from the proxy, plugins would have no way to prevent this.

**Comment by Sleepwalkerx** [comment]
> From my understanding, queue plugins have no way of knowing if a player got kicked from the backend. If this config option is turned on, plugins can still stop the proxy from disconnecting the player (and instead redirecting them) since such plugins already exist (KickRedirect). I'm just suggesting this because while redirect behavior plugins exist, there is no existing solution to disable the velocity redirect behavior for ALL kicks.

I think you can try to do something like this.
```kotlin
@Subscribe
    fun onPlayerKicked(event: KickedFromServerEvent) {
        val player = event.player
        val kickedFrom = event.server
        val current = player.currentServer.orElse(null)?.server
        if(current == null || current == kickedFrom) {
            val reason = event.serverKickReason.orElse(Component.text("Отключение от сервера"))
            event.result = KickedFromServerEvent.DisconnectPlayer.create(reason)
        }
    }
```
